src/gui_common/charts/line/LineChart.cs

```diff
@@ -736,6 +685,102 @@ private void FlattenLines(string datasetName)
         }
     }
 
+    /// <summary>
+    ///   Calculates the min/max values of this chart based on shown datasets and generates the axes scale ticks.
+    /// </summary>
+    private void UpdateMinimumAndMaximumValues()
+    {
+        // Default to zeros
+        MaxValues = Vector2.Zero;
+        MinValues = Vector2.Zero;
+
+        // Find the max values of all the data points first to make finding min values possible
+        foreach (var data in dataSets)
+        {
+            if (!data.Value.Draw)
+                continue;
+
+            foreach (var point in data.Value.DataPoints)
+            {
+                MaxValues = new Vector2(Mathf.Max(point.Value.x, MaxValues.x), Mathf.Max(point.Value.y, MaxValues.y));
+            }
+        }
+
+        MinValues = MaxValues;
+
+        // Find the minimum values of all the data points
+        foreach (var data in dataSets)
+        {
+            if (!data.Value.Draw)
+                continue;
+
+            foreach (var point in data.Value.DataPoints)
+            {
+                MinValues = new Vector2(Mathf.Min(point.Value.x, MinValues.x), Mathf.Min(point.Value.y, MinValues.y));
+            }
+        }
+
+        // If min/max turns out to be equal, set one of their point to zero as the initial value
+
+        if (MinValues.x == MaxValues.x)
+        {
+            if (MaxValues.x > 0)
+            {
+                MinValues = new Vector2(0, MinValues.y);
+            }
+            else if (MaxValues.x < 0)
+            {
+                MaxValues = new Vector2(0, MaxValues.y);
+            }
+        }
+
+        if (MinValues.y == MaxValues.y)
+        {
+            if (MaxValues.y > 0)
+            {
+                MinValues = new Vector2(MinValues.x, 0);
+            }
+            else if (MaxValues.y < 0)
+            {
+                MaxValues = new Vector2(MaxValues.x, 0);
+            }
+        }
+
+        horizontalLabelsContainer.QueueFreeChildren();
+        verticalLabelsContainer.QueueFreeChildren();
+
+        // Populate the rows
+        for (int i = 0; i < XAxisTicks; i++)
+        {
+            var label = new Label
+            {
+                SizeFlagsHorizontal = (int)SizeFlags.ExpandFill,
+                Align = Label.AlignEnum.Center,
+            };
+
+            label.Text = Math.Round(
+                i * (MaxValues.x - MinValues.x) / (XAxisTicks - 1) + MinValues.x, 1).FormatNumber();
+
+            horizontalLabelsContainer.AddChild(label);
+        }
+
+        // Populate the columns (in reverse order)
+        for (int i = YAxisTicks; i-- > 0;)
```

Comment:
I think this would look nicer if you separate the condition check and the i-- part (if possible).